doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -671,13 +671,15 @@ file is a particular case of module called *library file*.
       the time it was compiled.
 
 
-   .. exn:: Require is not allowed inside a module or a module type.
+   .. exn:: Require inside a module is deprecated and strongly discouraged. You can Require a
+            module at toplevel and optionally Import it inside another one.
+            [require-in-module,deprecated]
+      :undocumented:
 
       This command
-      is not allowed inside a module or a module type being defined. It is
-      meant to describe a dependency between compilation units. Note however
-      that the commands ``Import`` and ``Export`` alone can be used inside modules
-      (see Section :ref:`Import <import_qualid>`).
+      is not allowed inside a module or a module type. It is
+      meant to describe a dependency between compilation units. Note, however,
+      that the commands :cmd:`Import` and :cmd:`Export` can be used inside modules.
```

Comment:
You need to choose between removing the documentation, or adapting it to the fact that this is a warning rather than an error and in that case remove the `:undocumented:` option. BTW, maybe `:undocumented:` should trigger a (fatal) warning if it is provided but that there is still a body (that's not the first time I observe this).